src/main/java/fr/openent/competences/service/impl/DefaultMatiereService.java

```diff
@@ -354,6 +356,21 @@ public void getSousMatieres(String idMatiere, String idStructure, Handler<Either
         sql.prepared(query, params, SqlResult.validResultHandler(handler));
     }
 
+    public Future<JsonArray> getUnderSubjects(String idMatiere, String idStructure) {
+        Promise<JsonArray> underSubjectPromise = Promise.promise();
+        String query = " SELECT id_type_sousmatiere as id_sousmatiere, id_matiere , libelle " +
+                " FROM " + underSubjectTable + "  INNER JOIN " + typeUnderSubjectTable +
+                " ON id_type_sousmatiere = type_sousmatiere.id " +
+                " WHERE id_matiere = ? AND id_structure = ?";
+
+        JsonArray params = new JsonArray().add(idMatiere).add(idStructure);
+        sql.prepared(query, params, SqlResult.validResultHandler(FutureHelper.handlerJsonArray(underSubjectPromise,
+                "[DefaultMatiereService] : getUnderSubjects ")));
+
+        return underSubjectPromise.future();
+    }
```

Comment:
Même principe par rapport à mon commentaire au-dessus : utilise la méthode déjà existante pour polymorpher ta méthode `getUnderSubjects` que tu veux faire retourner une future

Exemple usage :

```suggestion
   public Future<JsonArray> getUnderSubjects(String idMatiere, String idStructure) {
        Promise<JsonArray> underSubjectPromise = Promise.promise();
        getSousMatieres(idMatiere, idStructure, event -> {
            if (event.isLeft()) {
                log.error(String.format("[Competences@%s::getUnderSubjects] Error during sql request: %s.",
                        this.getClass().getSimpleName(), event.left().getValue()));
                underSubjectPromise.fail(event.left().getValue());
            } else {
                underSubjectPromise.complete(event.right().getValue());
            }
        });

        return underSubjectPromise.future();
    }
```